File: 1/js/p1.js
const readline = require('readline');

const depths = [];
let count = 0;

const rl = readline.createInterface({
  input: process.stdin
});

rl.on('line', (l) => {
  depths.push(parseInt(l));
});

rl.on('close', () => {
  for (let i = 0; i < depths.length - 1; i++) {
    const a = depths[i];
    const b = depths[i + 1];

    if (b - a > 0) {
      count++;
    }
  }

  console.log(`Number of increases: ${count}`);
});

File: 1/lua/p1.lua
local depths = {}
local count = 0

for line in io.lines() do
  table.insert(depths, tonumber(line))
end

for i = 1, #depths - 1 do
  local a = depths[i]
  local b = depths[i + 1]

  if b - a > 0 then
    count = count + 1
  end
end

print('Number of increases', count)
